executor/restore_test.go

```diff
@@ -2,306 +2,270 @@ package executor
+				"HOSTNAME":           "",
+				"RESTIC_PASSWORD":    "",
+				"RESTIC_REPOSITORY":  "s3:http://localhost:9000/test-backend",
+				"RESTORE_S3ENDPOINT": "http://localhost:9000/test",
+				"STATS_URL":          "",
+			},
+			ExpectedSecretKeyRefs: map[string]string{
+				"AWS_ACCESS_KEY_ID":       "accessKey-backend",
+				"AWS_SECRET_ACCESS_KEY":   "secretKey-backend",
+				"RESTORE_ACCESSKEYID":     "accessKey",
+				"RESTORE_SECRETACCESSKEY": "secretKey",
+			},
+		},
+		"givenFolderRestoreResource_whenSetupEnvVars_expectCertainEnvVars": {
+			GivenJobConfig: newConfig(),
+			GivenResource:  newFolderRestoreResource(),
+			ExpectedEnvVars: map[string]string{
+				"AWS_ACCESS_KEY_ID":     "",
+				"AWS_SECRET_ACCESS_KEY": "",
+				"HOSTNAME":              "",
+				"RESTIC_PASSWORD":       "",
+				"RESTIC_REPOSITORY":     "s3:/",
+				"RESTORE_DIR":           "/restore",
+				"STATS_URL":             "",
+			},
+			ExpectedSecretKeyRefs: map[string]string{},
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			e := NewRestoreExecutor(*tt.GivenJobConfig)
+			envVars := e.setupEnvVars(tt.GivenResource)
+
+			actualEnvVars, actualSecretKeyRefs := extractVarsAndSecretRefs(envVars)
+
+			assert.Equal(t, actualEnvVars, tt.ExpectedEnvVars)
+			assert.Equal(t, actualSecretKeyRefs, tt.ExpectedSecretKeyRefs)
+		})
+	}
 }
-func volumeMountID(el interface{}) string {
-	v := el.(v1.VolumeMount)
-	return v.Name
+
+func extractVarsAndSecretRefs(envVars []corev1.EnvVar) (map[string]string, map[string]string) {
+	actualVars := make(map[string]string)
+	actualSecretRefs := make(map[string]string)
+	for _, envVar := range envVars {
+		if envVar.ValueFrom != nil && envVar.ValueFrom.SecretKeyRef != nil {
+			actualSecretRefs[envVar.Name] = envVar.ValueFrom.SecretKeyRef.Key
+		} else {
+			actualVars[envVar.Name] = envVar.Value
+		}
+	}
+	return actualVars, actualSecretRefs
 }
 
-func jobMatcher(restoreType string, additionalArgs []string, env Elements, volumes Elements, volumeMounts Elements) gtypes.GomegaMatcher {
-	additionalArgs = append([]string{"-restore"}, additionalArgs...)
-	additionalArgs = append(additionalArgs, "-restoreType", restoreType)
+func TestRestore_volumeConfig(t *testing.T) {
+	tests := map[string]struct {
+		GivenJobConfig      *job.Config
+		GivenResource       *k8upv1a1.Restore
+		ExpectedPVC         PVCExpectation
+		ExpectedVolumeMount VolumeMountExpectation
+	}{
+		"givenS3RestoreResource_whenVolumeConfig_expectNoPVCAndNoMount": {
+			GivenJobConfig: newConfig(),
+			GivenResource:  newS3RestoreResource(),
+		},
+		"givenFolderRestoreResource_whenVolumeConfig_expectPVCAndMount": {
+			GivenJobConfig: newConfig(),
+			GivenResource:  newFolderRestoreResource(),
+			ExpectedPVC: PVCExpectation{
+				Expected:  true,
+				ClaimName: "test",
+				ReadOnly:  false,
+			},
+			ExpectedVolumeMount: VolumeMountExpectation{
+				Expected: true,
+				Name:     "test",
+				Path:     "/restore",
+			},
```

Comment:
Maybe we can even remove the struct and assert directly in the test?
```suggestion
			ExpectedPVCClaimName: "test",
			ExpectedVolumeMountName: "test",
			ExpectedVolumeMountPath: "/restore"
```

Reply:
I find the structure cleaner, especially when passing them to `assertVolumeMounts` and `assertVolumes`.